components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -283,6 +343,9 @@ static void CANIO_tx_1kHz_PRD(void)
  */
 static void CANIO_tx_100Hz_PRD(void)
 {
+    if (cantx.tx_100Hz_msg < MSG_packTable_100Hz_SIZE) {
```

Comment:
should these check for non-zero?
```suggestion
    if ((cantx.tx_100Hz_msg > 0) && (cantx.tx_100Hz_msg < MSG_packTable_100Hz_SIZE)) {
```
?

Reply:
Yeah I think it would be better actually.

Reply:
Actually, I don't think this is necessary. From what I remember, those functions are called to reset the counters to zero. I've added those if statements to make sure the counter reached the packTable size (aka all the messages have been sent) before resetting the counter. So I don't really see the point of verifying for non-zeros.